lib/puppetserver/ca/action/generate.rb

```diff
@@ -87,16 +112,35 @@ def generate_root_and_intermediate_ca(settings, signing_digest)
           int_cert = sign_intermediate(root_key, root_cert, int_csr, valid_until, signing_digest)
           int_crl = create_crl_for(int_cert, int_key, valid_until, signing_digest)
 
+          master_key = host.create_private_key(settings[:keylength])
+          master_csr = host.create_csr(settings[:certname], master_key)
+          master_cert = sign_master_cert(int_key, int_cert, master_csr,
+                                       valid_until, signing_digest, subject_alt_names)
+
           FileSystem.ensure_dir(settings[:cadir])
+          FileSystem.ensure_dir(settings[:certdir])
+          FileSystem.ensure_dir(settings[:privatekeydir])
```

Comment:
Should we create the publickeydir/pubkeys too?

Reply:
Oh yeah, meant to look into that.

Reply:
Added.